src/hooks/useAllCities.js

```diff
@@ -1,28 +1,32 @@
-import { useMemo } from 'react';
+import { useEffect, useMemo } from 'react';
 import { useInfiniteQuery } from '@tanstack/react-query';
 import fetchCities from '../services/CitiesService';
 import transformCityItem from '../data/transformCityItem';
+import { useSearch } from '../context/SearchContext';
 
 const ALL_CITIES_QUERY_KEY = 'allCities';
 
 const useAllCities = ({ pageSize, sortOrder }) => {
+  const { searchQuery } = useSearch();
```

Comment:
Using `useSearch` hook here makes `useAllCities` hook highly coupled with the search feature. It will lead to the inability to use `useAllCities` hook outside the `SearchContext`.

This can be refactored to avoid this coupling.

Reply:
Yes, you are 100% right. With my current code, useAllCities cannot be used outside of the SearchContext context, which reduces its reusability and flexibility. I decided to pass the searchQuery as props to the useAllCities hook instead of getting it from the context. It should work.